docs/source/locality.rst

```diff
@@ -0,0 +1,80 @@
+Data Locality
+=============
+
+*Data movement often needlessly limits performance.*
+
+This is especially true for analytic computations.  ``Distributed`` minimizes
+data movement when possible and enables the user to take control when
+necessary.  This document describes current scheduling policies and user API
+around data locality.
+
+Current Policies
+----------------
+
+Task Submission
+```````````````
+
+In the common case distributed runs tasks on workers that already hold
+dependent data.  If you have a task ``f(x)`` that requires some data ``x`` then
+that task will very likely be run on the worker that already holds ``x``.
+
+If a task requires data split among multiple workers, then the scheduler chooses
+to run the task on the worker that requires the least data transfer to it.
+The size of each data element is measured by the workers using the
+``sys.getsizeof`` function, which depends on the ``__sizeof__`` protocol
+generally available on most relevant Python objects.
+
+Data Scatter
+````````````
+
+When a user scatters data from their local process to the distributed network
+this data is distributed in a round-robin fashion grouping by number of cores.
+So for example If we have two workers ``Alice`` and ``Bob``, each with two
```

Comment:
It is a nice idea and it would be doable to add.  We could also give them arbitrary aliases like `0, 1, 2, ...` by default.